translator/jsonconfig/mergeJsonConfig.go

```diff
@@ -25,11 +25,12 @@ func MergeJsonConfigMaps(jsonConfigMapMap map[string]map[string]interface{}, def
 				return util.GetJsonMapFromJsonBytes([]byte(config.DefaultECSJsonConfig()))
 			}
 		}
-		if multiConfig == "remove" {
-			fmt.Println("No json config files found, please provide config, exit now")
+		
+		if strictValidation || multiConfig == "remove" {
```

Comment:
Personally, I think we should use a different exit code, and thus a different conditional branch for when the JSON fails validation - a failed JSON configuration translation is not the same as missing a JSON file in my opinion.

Reply:
Cause the current code is exit after detecting there is no json configuration was founding (since the json is invalid); therefore, exiting with `ERR_CODE_NOJSONFILE` . However, I got stuck between [exit after founding there is no json configuration](https://github.com/aws/amazon-cloudwatch-agent/pull/412/files#diff-e64720159e8e883ef2291f9df929db8abbc5901003168fc2568090ad638addfdR31) because of invalid json and [during the invalidation](https://github.com/aws/amazon-cloudwatch-agent/pull/412/files#diff-eb5d801c3c6e31c5d7ef534897e7e095a1f6344597db7d95adce51d962ec1038R186-R200) because of how we handle error. However, fixing it after some thoughts again.

Reply:
After thinking some thoughts again, the ```fmt.Println("No json config files found, please provide config, exit now")``` makes me thinking of **we only output this error when the json is missing**. However, that's not true since if that json configuration was not found, it would error before coming to this condition (image 1). Therefore, the only possible errors that come to this condition would be `invalid json configuration`. So changed the error from `ERR_CODE_NOJSONFILE` to `ERR_CODE_INVALIDJSONFILE` if that makes sense.
![image](https://user-images.githubusercontent.com/91758108/160142453-2116eb21-c668-49d2-908e-626452500e51.png)

Reply:
When we exit, can we not distinguish between there being an error parsing a JSON file versus no JSON file being supplied? At the end of the day, I think this change is fine because technically having no JSON file is "invalid".

Reply:
Agree on later comments. The way we handling error right now is: checking if file exist from [config download] (https://github.com/aws/amazon-cloudwatch-agent/blob/master/cmd/config-downloader/downloader.go#L195-L217) -> error parsing a JSON file from [config translator]. Therefore, yes we can handle an error code for parsing JSON file and no JSON file being supplied. However, this still does not change anything much though. `config-translator` is only for translator, not fetching so the `fmt.Println("No json config files found, please provide config, exit now")` does not make sense to me.

Reply:
Does this makes output become "invalid json configuration" when customer doesn't provide a json config at all? Still think we should distinguish missing file vs invalid file.

Reply:
I would say no json file is invalid json file, but I do agree it would be better to separate.

Reply:
If that's the case, change how we handle errors a little bit. There would be lots of scenario but able to write the final code for these scenarios.